### ISSUE TITLE
Essential packages were removed and -y was used without --allow-remove-essential

### ISSUE THREAD

**Issue by kpcyrd** [fepitre/debrebuild#8]
hi!

I noticed some builds fail with this error:

```
+ /debrebuild/debrebuild.py --output=/tmp/rebuilderdu5zMdm/out --builder=mmdebstrap -- /tmp/rebuilderdu5zMdm/inputs/xwatch_2.11-16_amd64.buildinfo
[...]
The following packages will be upgraded:
  base-files base-passwd bash bsdutils coreutils dash debconf debianutils
  findutils grep gzip hostname init-system-helpers libacl1 libattr1
  libaudit-common libaudit1 libblkid1 libbz2-1.0 libc-bin libc6 libcap-ng0
  libdb5.3 libdebconfclient0 libfdisk1 libgcrypt20 libgmp10 libgpg-error0
  liblz4-1 liblzma5 libmount1 libpcre3 libseccomp2 libselinux1 libsmartcols1
  libstdc++6 libsystemd0 libtinfo6 libudev1 libunistring2 libuuid1 login mawk
  ncurses-base ncurses-bin perl-base sysvinit-utils tar util-linux zlib1g
WARNING: The following essential packages will be removed.
This should NOT be done unless you know exactly what you are doing!
  fdisk libncursesw6 (due to fdisk)
50 upgraded, 103 newly installed, 2 to remove and 18 not upgraded.
E: Essential packages were removed and -y was used without --allow-remove-essential.
E: run_chroot failed: E: env --unset=APT_CONFIG --unset=TMPDIR /usr/sbin/chroot /tmp/mmdebstrap.qrxgK8mUSg apt-get --yes install -oAPT::Status-Fd=<$fd> -oDpkg::Use-Pty=false base-files:amd64=11 xorg-sgml-doctools:all=1:1.11-1 dash:amd64=0.5.10.2-7 
[...]
```

I think `--allow-remove-essential` was recently introduced in apt and needs to be added to debrebuild.py.

**Full log:**

<details>

```
+ /debrebuild/debrebuild.py --output=/tmp/rebuilderdu5zMdm/out --builder=mmdebstrap -- /tmp/rebuilderdu5zMdm/inputs/xwatch_2.11-16_amd64.buildinfo
I: automatically chosen mode: root
I: chroot architecture amd64 is equal to the host's architecture
I: automatically chosen format: tar
I: using /tmp/mmdebstrap.qrxgK8mUSg as tempdir
I: running apt-get update...
I: downloading packages with apt...
I: extracting archives...
I: installing essential packages...
I: running --essential-hook in shell: sh -c 'chroot "$1" sh -c "apt-get --yes install fakeroot util-linux"' exec /tmp/mmdebstrap.qrxgK8mUSg
Reading package lists...
Building dependency tree...
util-linux is already the newest version (2.33.1-0.1).
The following NEW packages will be installed:
  fakeroot libfakeroot
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 132 kB of archives.
After this operation, 373 kB of additional disk space will be used.
Get:1 http://snapshot.notset.fr/archive/debian/20190706T233220Z bullseye/main amd64 libfakeroot amd64 1.23-1 [45.9 kB]
Get:2 http://snapshot.notset.fr/archive/debian/20190706T233220Z bullseye/main amd64 fakeroot amd64 1.23-1 [85.8 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 132 kB in 1s (212 kB/s)
Selecting previously unselected package libfakeroot:amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 4453 files and directories currently installed.)
Preparing to unpack .../libfakeroot_1.23-1_amd64.deb ...
Unpacking libfakeroot:amd64 (1.23-1) ...
Selecting previously unselected package fakeroot.
Preparing to unpack .../fakeroot_1.23-1_amd64.deb ...
Unpacking fakeroot (1.23-1) ...
Setting up libfakeroot:amd64 (1.23-1) ...
Setting up fakeroot (1.23-1) ...
update-alternatives: using /usr/bin/fakeroot-sysv to provide /usr/bin/fakeroot (fakeroot) in auto mode
Processing triggers for libc-bin (2.28-10) ...
I: running special hook: copy-in /usr/share/keyrings/debian-archive-bullseye-automatic.gpg /usr/share/keyrings/debian-archive-bullseye-security-automatic.gpg /usr/share/keyrings/debian-archive-bullseye-stable.gpg /usr/share/keyrings/debian-archive-buster-automatic.gpg /usr/share/keyrings/debian-archive-buster-security-automatic.gpg /usr/share/keyrings/debian-archive-buster-stable.gpg /usr/share/keyrings/debian-archive-keyring.gpg /usr/share/keyrings/debian-archive-removed-keys.gpg /usr/share/keyrings/debian-archive-stretch-automatic.gpg /usr/share/keyrings/debian-archive-stretch-security-automatic.gpg /usr/share/keyrings/debian-archive-stretch-stable.gpg /usr/share/keyrings/debian-ports-archive-keyring-removed.gpg /usr/share/keyrings/debian-ports-archive-keyring.gpg /usr/share/keyrings/debian-keyring.gpg /etc/apt/trusted.gpg.d/
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-bullseye-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-bullseye-security-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-bullseye-stable.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-buster-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-buster-security-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-buster-stable.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-keyring.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-removed-keys.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-stretch-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-stretch-security-automatic.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-archive-stretch-stable.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-ports-archive-keyring-removed.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-ports-archive-keyring.gpg': Operation not supported
tar: setxattrat: Cannot set 'security.selinux' extended attribute for file 'debian-keyring.gpg': Operation not supported
I: running --essential-hook in shell: sh -c 'chroot "$1" sh -c "rm /etc/apt/sources.list && echo 'deb http://snapshot.notset.fr/archive/debian/20210814T212851Z/ bookworm main
deb-src http://snapshot.notset.fr/archive/debian/20210814T212851Z/ bookworm main
deb http://snapshot.notset.fr/archive/debian/20190706T233220Z/ bullseye main
deb http://snapshot.notset.fr/archive/debian/20200530T025521Z/ bullseye main
deb http://snapshot.notset.fr/archive/debian/20200603T091236Z/ bullseye main
deb http://snapshot.notset.fr/archive/debian/20200525T145753Z/ unstable main' >> /etc/apt/sources.list && apt-get update"' exec /tmp/mmdebstrap.qrxgK8mUSg
Get:1 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm InRelease [81.6 kB]
Hit:2 http://snapshot.notset.fr/archive/debian/20190706T233220Z bullseye InRelease
Get:3 http://snapshot.notset.fr/archive/debian/20200530T025521Z bullseye InRelease [116 kB]
Ign:4 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main Sources
Get:5 http://snapshot.notset.fr/archive/debian/20200603T091236Z bullseye InRelease [116 kB]
Ign:6 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main amd64 Packages
Ign:4 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main Sources
Get:7 http://snapshot.notset.fr/archive/debian/20200525T145753Z unstable InRelease [146 kB]
Ign:6 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main amd64 Packages
Ign:8 http://snapshot.notset.fr/archive/debian/20200530T025521Z bullseye/main amd64 Packages
Ign:4 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main Sources
Ign:6 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main amd64 Packages
Ign:8 http://snapshot.notset.fr/archive/debian/20200530T025521Z bullseye/main amd64 Packages
Ign:9 http://snapshot.notset.fr/archive/debian/20200603T091236Z bullseye/main amd64 Packages
Get:4 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main Sources [11.4 MB]
Get:6 http://snapshot.notset.fr/archive/debian/20210814T212851Z bookworm/main amd64 Packages [11.1 MB]
Ign:10 http://snapshot.notset.fr/archive/debian/20200525T145753Z unstable/main amd64 Packages
Ign:8 http://snapshot.notset.fr/archive/debian/20200530T025521Z bullseye/main amd64 Packages
Ign:9 http://snapshot.notset.fr/archive/debian/20200603T091236Z bullseye/main amd64 Packages
Ign:10 http://snapshot.notset.fr/archive/debian/20200525T145753Z unstable/main amd64 Packages
Get:8 http://snapshot.notset.fr/archive/debian/20200530T025521Z bullseye/main amd64 Packages [10.5 MB]
Ign:9 http://snapshot.notset.fr/archive/debian/20200603T091236Z bullseye/main amd64 Packages
Ign:10 http://snapshot.notset.fr/archive/debian/20200525T145753Z unstable/main amd64 Packages
Get:9 http://snapshot.notset.fr/archive/debian/20200603T091236Z bullseye/main amd64 Packages [10.5 MB]
Get:10 http://snapshot.notset.fr/archive/debian/20200525T145753Z unstable/main amd64 Packages [11.2 MB]
Fetched 55.0 MB in 46s (1187 kB/s)
Reading package lists...
I: installing remaining packages inside the chroot...
Reading package lists...
Building dependency tree...
Reading state information...
sed is already the newest version (4.7-1).
diffutils is already the newest version (1:3.7-3).
dpkg is already the newest version (1.19.7).
libpam-modules is already the newest version (1.3.1-5).
libpam-modules-bin is already the newest version (1.3.1-5).
libpam-runtime is already the newest version (1.3.1-5).
libpam0g is already the newest version (1.3.1-5).
The following additional packages will be installed:
  libfdisk1
Suggested packages:
  autoconf-archive gnu-standards autoconf-doc bash-doc binutils-doc wamerican
  | wordlist whois vacation bzip2-doc cpp-doc gcc-9-locales debconf-doc
  debconf-utils whiptail | dialog libterm-readline-gnu-perl libgtk3-perl
  libnet-ldap-perl debconf-kde-helper dh-make debian-keyring mlocate | locate
  g++-multilib g++-9-multilib gcc-9-doc gcc-multilib manpages-dev flex bison
  gdb gcc-doc gcc-9-multilib gettext-doc libasprintf-dev libgettextpo-dev
  groff less glibc-doc libc-l10n locales gnupg | gnupg2 git bzr libforms-bin
  libforms-doc rng-tools gdbm-l10n libstdc++-9-doc libtool-doc gfortran
  | fortran95-compiler gcj-jdk libx11-doc libxcb-doc m4-doc make-doc apparmor
  www-browser ed diffutils-doc perl-doc libterm-readline-gnu-perl
  | libterm-readline-perl-perl libb-debug-perl liblocale-codes-perl
  libtap-harness-archive-perl libmail-box-perl ncompress tar-scripts tar-doc
  dosfstools kbd | console-tools util-linux-locales yodl-doc
Recommended packages:
  bash-completion apt-utils debconf-i18n gnupg | gnupg2
  libalgorithm-merge-perl curl | wget | lynx manpages manpages-dev
  libfile-fcntllock-perl liblocale-gettext-perl libarchive-cpio-perl
  libglib2.0-data shared-mime-info xdg-user-dirs libgpg-error-l10n libltdl-dev
  uuid-runtime netbase libmail-sendmail-perl
The following packages will be REMOVED:
  fdisk libncursesw6
The following NEW packages will be installed:
  autoconf automake autopoint autotools-dev binutils binutils-common
  binutils-x86-64-linux-gnu bsdmainutils build-essential bzip2 cpp cpp-9
  debhelper dh-autoreconf dh-strip-nondeterminism dpkg-dev dwz file g++ g++-9
  gcc gcc-10-base gcc-9 gcc-9-base gettext gettext-base groff-base
  intltool-debian libarchive-zip-perl libasan5 libatomic1 libbinutils libbsd0
  libc-dev-bin libc6-dev libcc1-0 libcroco3 libcrypt-dev libcrypt1
  libctf-nobfd0 libctf0 libdebhelper-perl libdpkg-perl libelf1 libffi7
  libfile-stripnondeterminism-perl libforms-dev libforms2 libgcc-9-dev
  libgcc-s1 libgdbm-compat4 libgdbm6 libglib2.0-0 libgomp1 libicu63 libisl22
  libitm1 libjpeg62-turbo liblsan0 libmagic-mgc libmagic1 libmpc3 libmpfr6
  libpcre2-8-0 libperl5.30 libpipeline1 libpthread-stubs0-dev libquadmath0
  libsigsegv2 libstdc++-9-dev libsub-override-perl libtool libtsan0 libubsan1
  libuchardet0 libx11-6 libx11-data libx11-dev libxau-dev libxau6 libxcb1
  libxcb1-dev libxdmcp-dev libxdmcp6 libxml2 libxpm-dev libxpm4 linux-libc-dev
  lsb-base m4 make man-db patch perl perl-modules-5.30 po-debconf
  sensible-utils x11proto-core-dev x11proto-dev xorg-sgml-doctools xtrans-dev
  xz-utils yodl
The following packages will be upgraded:
  base-files base-passwd bash bsdutils coreutils dash debconf debianutils
  findutils grep gzip hostname init-system-helpers libacl1 libattr1
  libaudit-common libaudit1 libblkid1 libbz2-1.0 libc-bin libc6 libcap-ng0
  libdb5.3 libdebconfclient0 libfdisk1 libgcrypt20 libgmp10 libgpg-error0
  liblz4-1 liblzma5 libmount1 libpcre3 libseccomp2 libselinux1 libsmartcols1
  libstdc++6 libsystemd0 libtinfo6 libudev1 libunistring2 libuuid1 login mawk
  ncurses-base ncurses-bin perl-base sysvinit-utils tar util-linux zlib1g
WARNING: The following essential packages will be removed.
This should NOT be done unless you know exactly what you are doing!
  fdisk libncursesw6 (due to fdisk)
50 upgraded, 103 newly installed, 2 to remove and 18 not upgraded.
E: Essential packages were removed and -y was used without --allow-remove-essential.
E: run_chroot failed: E: env --unset=APT_CONFIG --unset=TMPDIR /usr/sbin/chroot /tmp/mmdebstrap.qrxgK8mUSg apt-get --yes install -oAPT::Status-Fd=<$fd> -oDpkg::Use-Pty=false base-files:amd64=11 xorg-sgml-doctools:all=1:1.11-1 dash:amd64=0.5.10.2-7 libbinutils:amd64=2.34-8 libquadmath0:amd64=10.1.0-3 binutils-x86-64-linux-gnu:amd64=2.34-8 libpam0g:amd64=1.3.1-5 debconf:all=1.5.74 man-db:amd64=2.9.1-1 libxcb1:amd64=1.14-2 libitm1:amd64=10.1.0-3 bash:amd64=5.0-6 libc-dev-bin:amd64=2.30-8 libarchive-zip-perl:all=1.68-1 lsb-base:all=11.1.0 gcc:amd64=4:9.2.1-3.1 gettext:amd64=0.19.8.1-10 libaudit-common:all=1:2.8.5-3 libudev1:amd64=245.5-3 libgdbm-compat4:amd64=1.18.1-5 autotools-dev:all=20180224.1 libicu63:amd64=63.2-3 libubsan1:amd64=10.1.0-3 libsystemd0:amd64=245.5-3 util-linux:amd64=2.35.2-2 libblkid1:amd64=2.35.2-2 libxpm4:amd64=1:3.5.12-1 libsub-override-perl:all=0.09-2 libcroco3:amd64=0.6.13-1 base-passwd:amd64=3.5.47 libjpeg62-turbo:amd64=1:1.5.2-2+b1 ncurses-bin:amd64=6.2-1 libgcrypt20:amd64=1.8.5-5 libc6:amd64=2.30-8 grep:amd64=3.4-1 gcc-10-base:amd64=10.1.0-3 sensible-utils:all=0.0.12+nmu1 libtsan0:amd64=10.1.0-3 libdb5.3:amd64=5.3.28+dfsg1-0.6 autopoint:all=0.19.8.1-10 x11proto-core-dev:all=2020.1-1 libcrypt1:amd64=1:4.4.16-1 libx11-6:amd64=2:1.6.9-2+b1 libgpg-error0:amd64=1.37-1 cpp-9:amd64=9.3.0-13 bzip2:amd64=1.0.8-3 gettext-base:amd64=0.19.8.1-10 groff-base:amd64=1.22.4-5 init-system-helpers:all=1.57 cpp:amd64=4:9.2.1-3.1 libsmartcols1:amd64=2.35.2-2 x11proto-dev:all=2020.1-1 dpkg-dev:all=1.19.7 liblzma5:amd64=5.2.4-1+b1 bsdmainutils:amd64=11.1.2+b1 libxpm-dev:amd64=1:3.5.12-1 libtool:all=2.4.6-14 libffi7:amd64=3.3-4 libpipeline1:amd64=1.5.2-2 libselinux1:amd64=3.0-1+b3 diffutils:amd64=1:3.7-3 libxau-dev:amd64=1:1.0.8-1+b2 autoconf:all=2.69-11.1 libgdbm6:amd64=1.18.1-5 bsdutils:amd64=1:2.35.2-2 gzip:amd64=1.10-2 libperl5.30:amd64=5.30.2-1 libpcre3:amd64=2:8.39-12+b1 hostname:amd64=3.23 libglib2.0-0:amd64=2.64.2-1 po-debconf:all=1.0.21 libsigsegv2:amd64=2.12-2 libxau6:amd64=1:1.0.8-1+b2 gcc-9:amd64=9.3.0-13 libxml2:amd64=2.9.10+dfsg-5 libpam-modules-bin:amd64=1.3.1-5 intltool-debian:all=0.35.0+20060710.5 make:amd64=4.3-1 dpkg:amd64=1.19.7 libunistring2:amd64=0.9.10-2 libctf-nobfd0:amd64=2.34-8 libattr1:amd64=1:2.4.48-5 libgcc-9-dev:amd64=9.3.0-13 binutils-common:amd64=2.34-8 perl:amd64=5.30.2-1 debhelper:all=13.1 libfile-stripnondeterminism-perl:all=1.8.1-1 g++:amd64=4:9.2.1-3.1 libuchardet0:amd64=0.0.6-3 libmount1:amd64=2.35.2-2 libmagic-mgc:amd64=1:5.38-5 libgomp1:amd64=10.1.0-3 tar:amd64=1.30+dfsg-7 libgmp10:amd64=2:6.2.0+dfsg-4 login:amd64=1:4.8.1-1 libforms-dev:amd64=1.2.3-1.4 libdpkg-perl:all=1.19.7 file:amd64=1:5.38-5 libcap-ng0:amd64=0.7.9-2.1+b2 dh-strip-nondeterminism:all=1.8.1-1 libbz2-1.0:amd64=1.0.8-3 libforms2:amd64=1.2.3-1.4 coreutils:amd64=8.30-3+b1 libuuid1:amd64=2.35.2-2 libaudit1:amd64=1:2.8.5-3+b1 libelf1:amd64=0.176-1.1 libc6-dev:amd64=2.30-8 gcc-9-base:amd64=9.3.0-13 libasan5:amd64=9.3.0-13 libstdc++-9-dev:amd64=9.3.0-13 libcc1-0:amd64=10.1.0-3 libdebconfclient0:amd64=0.252 libcrypt-dev:amd64=1:4.4.16-1 libc-bin:amd64=2.30-8 libctf0:amd64=2.34-8 perl-modules-5.30:all=5.30.2-1 mawk:amd64=1.3.4.20200120-2 libstdc++6:amd64=10.1.0-3 sysvinit-utils:amd64=2.96-3 libatomic1:amd64=10.1.0-3 perl-base:amd64=5.30.2-1 libseccomp2:amd64=2.4.3-1+b1 sed:amd64=4.7-1 libgcc-s1:amd64=10.1.0-3 zlib1g:amd64=1:1.2.11.dfsg-2 libxdmcp6:amd64=1:1.1.2-3 libpthread-stubs0-dev:amd64=0.4-1 libmpfr6:amd64=4.0.2-1 libmagic1:amd64=1:5.38-5 m4:amd64=1.4.18-4 dwz:amd64=0.13-5 libmpc3:amd64=1.1.0-1 dh-autoreconf:all=19 libx11-dev:amd64=2:1.6.9-2+b1 xtrans-dev:all=1.4.0-1 libxcb1-dev:amd64=1.14-2 patch:amd64=2.7.6-6 libxdmcp-dev:amd64=1:1.1.2-3 libacl1:amd64=2.2.53-8 linux-libc-dev:amd64=5.6.14-1 ncurses-base:all=6.2-1 libpam-modules:amd64=1.3.1-5 build-essential:amd64=12.8 liblz4-1:amd64=1.9.2-2 yodl:amd64=4.02.02-2 libisl22:amd64=0.22.1-1 xz-utils:amd64=5.2.4-1+b1 binutils:amd64=2.34-8 debianutils:amd64=4.11 libdebhelper-perl:all=13.1 libx11-data:all=2:1.6.9-2 liblsan0:amd64=10.1.0-3 automake:all=1:1.16.2-1 libbsd0:amd64=0.10.0-1 findutils:amd64=4.7.0-1 libtinfo6:amd64=6.2-1 g++-9:amd64=9.3.0-13 libpcre2-8-0:amd64=10.34-7 libpam-runtime:all=1.3.1-5 failed
W: listening on child socket failed: 
I: removing tempdir /tmp/mmdebstrap.qrxgK8mUSg...
mmdebstrap failed
```

</details>

**Comment by fepitre** [comment]
I encountered this exact transaction issue multiple times for which I already mitigated it on several cases in the past...It looks like this one is again appearing. Allowing to remove essential simply breaks things and this is why it's not added. I'll try to improve that again but also looking at simply installing every single package version only. I think josch did something similar with mmdebstrap for debrebuild.pl.
